code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -918,111 +922,100 @@ pub mod pallet {
 			Ok(())
 		}
 
+		// TODO(benluelo): Split this out into a separate function/file
 		#[transactional]
 		fn split(
 			who: &Self::AccountId,
-			(fnft_collection_id, fnft_instance_id): &Self::PositionId,
+			(fnft_collection_id, existing_fnft_instance_id): &Self::PositionId,
 			ratio: Permill,
-		) -> Result<[Self::PositionId; 2], DispatchError> {
-			// TODO(benluelo): Refactor
-			let keep_alive = false;
-			let mut old_position_stake = BalanceOf::<T>::zero();
-			let mut old_position =
-				Stakes::<T>::try_mutate(fnft_collection_id, fnft_instance_id, |maybe_stake| {
-					let stake = maybe_stake.as_mut().ok_or(Error::<T>::StakeNotFound)?;
-
-					let old_value = stake.clone();
-					stake.stake = ratio.mul_floor(stake.stake);
-					stake.share = ratio.mul_floor(stake.share);
+		) -> Result<Self::PositionId, DispatchError> {
+			let (new_fnft_instance_id, new_position) = Stakes::<T>::try_mutate(
+				fnft_collection_id,
+				existing_fnft_instance_id,
+				|maybe_old_position| {
+					let existing_position =
+						maybe_old_position.as_mut().ok_or(Error::<T>::StakeNotFound)?;
+
+					let left_from_one_ratio = ratio.left_from_one();
+
+					// create the new position first, before mutating the old position
+					// mul_ceil is used for the new position, and mul_floor for the old position,
+					// that way any rounding is accounted for.
+					let new_stake = left_from_one_ratio.mul_ceil(existing_position.stake);
```

Comment:
We need a failing test that captures the problem fixed by doing floor/ceil

Reply:
https://github.com/ComposableFi/composable/commit/913dcfc8974ec2e1acb355eba5634e85d3b3dc70